src/main/kotlin/com/atlassian/performance/tools/report/api/ReportStrategy.kt

```diff
@@ -0,0 +1,25 @@
+package com.atlassian.performance.tools.report.api
+
+import com.atlassian.performance.tools.jiraactions.api.ActionMetricStatistics
+
+object ReportStrategy {
+    class CombinedReport(actionStats: ActionMetricStatistics): Report {
+        private val csv = CSVReport(actionStats)
+        private val plain = PlaintextReport(actionStats)
+
+        override fun generate(): String {
+            return plain.generate() + csv.generate()
+        }
+    }
+
+    fun newReport(actionStats: ActionMetricStatistics): Report {
+        // declared locally as value may change under test
+        val outputFormat = System.getProperty("report.format") ?: "plain"
```

Comment:
That's an unnecessary and dangerous precedence.
We got Java API, so Java programs can build benchmarks. Java allows for concurrency, which is powerful and incredibly useful (e.g. [multi-hardware explorations](https://github.com/atlassian/jira-hardware-exploration)). Controlling the benchmark behavior with system properties adds a global state, which forbids concurrency.

We gotta expose APIs, which allow *local* control of the features. E.g. pass it as a parameter, or just let the caller call the reports they need.

Reply:
Or let's just produce both reports (plaintext + csv) by default and don't introduce any configuration (less code, less complexity added).

Reply:
@dagguh do the recent changes resolve the issue here?

Reply:
All good 👍